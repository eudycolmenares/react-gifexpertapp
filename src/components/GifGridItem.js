import PropTypes from 'prop-types'
// 
const GifGridItem = ({ title, url }) => {
    // console.log(image);
    return (
        <div className="card animate__animated animate__zoomIn animate__faster">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
// 
export default GifGridItem