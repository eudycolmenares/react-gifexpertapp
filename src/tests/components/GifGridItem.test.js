import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'
// 
describe('Test in <GifGridItem />', () => {
    const title = 'Titulo!'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )
    test('Should display the component correctly!', () => {
        expect( wrapper ).toMatchSnapshot()
    })
    test('Shoud have a paragraph with the title', () => {
        const p = wrapper.find( 'p' )
        expect( p.text() ).toBe( title )
    })
    test('Shoud have a image like data from props!', () => {
        const img = wrapper.find( 'img' )
        expect( img.prop('src') ).toBe( url )
        expect( img.prop('alt') ).toBe( title )
    })
    test('Shoud have class "animate__zoomIn"', () => {
        const div = wrapper.find( 'div' )
        const className = div.prop('className')
        expect( className.includes('animate__zoomIn') ).toBe( true )
    })
})