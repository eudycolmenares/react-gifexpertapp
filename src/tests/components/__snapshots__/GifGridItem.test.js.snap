// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> Should display the component correctly! 1`] = `
<div
  className="card animate__animated animate__zoomIn animate__faster"
>
  <img
    alt="Titulo!"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo!
  </p>
</div>
`;
